api/rds.proto

```diff
@@ -585,6 +588,22 @@ message RedirectAction {
   RedirectResponseCode response_code = 3 [(validate.rules).enum.defined_only = true];
 }
 
+message DirectResponseAction {
+  // Specifies the HTTP response status to be returned.
+  google.protobuf.UInt32Value status = 1 [(validate.rules).message.required = true];
+
+  // Specifies the human-readable status to be returned if
+  // the response is encoded as HTTP/1.x.
+  string reason = 2;
+
+  // Specifies the response content-type. This setting defaults to text/plain.
+  string content_type = 3;
+
+  // Specifies the content of the response body. If this setting is omitted,
+  // no body is included in the generated response.
+  string body = 4;
```

Comment:
I think this should be a DataSource, so that it can either be a file on disk, or provided inline.

Reply:
+1 good idea.

Reply:
There's a practical problem with DataSource: its inline value is of type `bytes`, which will require people using JSON configs to [base64-encode](https://developers.google.com/protocol-buffers/docs/proto3#json) the value.

Do you have any objection to me extending DataSource to support a `string` option, to save JSON users the trouble of base64-encoding?

Reply:
+1, better if we can make it use one_of, which I think is wire compatible but @htuch and @wora will need to confirm.

Reply:
@mattklein123 I don't understand.  DataSource is already one_of ( bytes, filepath ).  What are you wanting to make one_of?  I think @brian-pane was asking if he can add another field to the existing one-of so it has (bytes, filepath, string).  

If I'm understanding correctly, I have no objection to that, but I don't know if there's any negative protobuf side-effects, such as other uses of DataSource having to check for another value in the one-of.

If I'm not understanding correctly, can someone clarify for me?

Reply:
Yes, I'm proposing to add a string to the existing one_of.

Reply:
Sorry I didn't look at DataSource, so didn't know if it already had one_of. Yes, go for it! :)

Reply:
Yes, this is fine. BTW, I'm moving `DataSource` around to its own distinct file in #398.

Reply:
I'm happy to split out that part of the PR if it helps speed along this one.